EarlGrey/Action/GREYActions.m

```diff
@@ -262,6 +266,31 @@ @implementation GREYActions
 #pragma mark - Private
 
 /**
+ *  Set the UITextField text value directly, bypassing the iOS keyboard.
+ *  @param text     The text to be typed.
+ *
+ *  @return @c YES if the action succeeded, else @c NO. If an action returns @c NO, it does not
+ *          mean that the action was not performed at all but somewhere during the action execution
+ *          the error occured and so the UI may be in an unrecoverable state.
+ */
++ (id<GREYAction>)grey_actionForReplaceText:(NSString *)text {
+    return [GREYActionBlock actionWithName:[NSString stringWithFormat:@"Type \"%@\"", text]
+                               constraints:grey_not(grey_systemAlertViewShown())
+                              performBlock:^BOOL (id element, __strong NSError **errorOrNil) {
+      if (![element isKindOfClass:[UITextField class]]) {
```

Comment:
The code should probably be updated to support webviews as well (which can't be done if the constraint is UITextField).